pkg/cli/democluster/demo_cluster.go

```diff
@@ -228,10 +228,11 @@ func (c *transientCluster) Start(
 	// Step 1: create the first node.
 	{
 		phaseCtx := logtags.AddTag(ctx, "phase", 1)
-		c.infoLog(phaseCtx, "creating the first node")
+		ctx := phaseCtx // shadow the original ctx; folk prefer this name to "phaseCtx".
```

Comment:
nit (for all): `folk prefer this name to "phaseCtx"` didn't make sense to me as to what class of error this was preventing until i read the commit message. maybe `shadow the original ctx to prevent accidentally using the original ctx instead of the phaseCtx which can be a easy reach as a mistake` or something?

alternatively, make all these steps individual functions passing in the phase context and having the function take in `ctx` and avoid the shadowing entirely. hopefully the refactor for that isn't too nasty.

Reply:
yeah, the individual functions seems to be the better choice. I'll work on that next.

Reply:
(the linter doesn't like the shadowing anyway)